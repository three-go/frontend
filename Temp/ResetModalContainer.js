import React, { useState } from "react";

import { useNavigation } from "@react-navigation/native";

import { ButtonModal, SmallButton } from "../src/components";

const ResetModalContainer = () => {
  const [modalVisible, setModalVisible] = useState(true);
  const navigation = useNavigation();

  const handleNavigateMain = () => {
    console.log("초기화 로직 필요?");
    navigation.navigate("Main");
    setModalVisible(false);
  };

  return (
    <ButtonModal isVisible={modalVisible} content="남은 하트가 없습니다.">
      <SmallButton
        content="메인으로"
        color="#c92a2a"
        onPress={handleNavigateMain}
      />
    </ButtonModal>
  );
};

export default ResetModalContainer;
