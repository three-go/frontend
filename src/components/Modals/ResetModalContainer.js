import React, { useState } from "react";

import { useNavigation } from "@react-navigation/native";

import { ButtonModal, SmallButton } from "..";

const ResetModalContainer = () => {
  const [modalVisible, setModalVisible] = useState(true);
  const navigation = useNavigation();

  const handleNavigateMain = () => {
    navigation.navigate("Main");
    setModalVisible(false);
  };

  return (
    <ButtonModal isVisible={modalVisible} content="남은 하트가 없습니다.">
      <SmallButton
        content="메인으로"
        color="#c92a2a"
        onPress={handleNavigateMain}
      />
    </ButtonModal>
  );
};

export default ResetModalContainer;
