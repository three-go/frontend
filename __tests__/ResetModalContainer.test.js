import React from "react";

import { render, fireEvent } from "@testing-library/react-native";

import ResetModalContainer from "../src/components/Modals/ResetModalContainer";

const mockNavigate = jest.fn();

jest.mock("@react-navigation/native", () => {
  return {
    useNavigation: () => ({
      navigate: mockNavigate,
    }),
  };
});

describe("ResetModalContainer", () => {
  it("ResetModalContainer render check", () => {
    const component = render(<ResetModalContainer />);

    expect(component.getByText("남은 하트가 없습니다.")).toBeTruthy();
    expect(component.getByText("메인으로")).toBeTruthy();
    expect(component.toJSON).toMatchSnapshot();
    expect(component.toJSON).toBeTruthy();
  });

  it("ResetModalContainer '메인으로' button click", () => {
    const component = render(<ResetModalContainer />);

    const mainNavigateButton = component.getByText("메인으로");
    fireEvent.press(mainNavigateButton);

    expect(mockNavigate.mock.calls.length).toBe(1);
    expect(() => component.getByText("메인으로")).toThrow(
      "Unable to find an element with text: 메인으로"
    );
  });
});
