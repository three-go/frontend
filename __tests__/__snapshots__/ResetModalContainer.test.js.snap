// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResetModalContainer ResetModalContainer render check 1`] = `[Function]`;
